### PR TITLE
Varonis saas dev

### PR DESCRIPTION
**Pull request by vkorenkov-varonis**
Required items, please complete
   
   Change(s):
   - Add new solution that includes connection to Varonis DatAlert service and workbook

   Reason for Change(s):
   - create new data connector that ingest data from Varonis DatAlert service

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - Need help

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @vkorenkov-varonis, 

1. Please resolve the validation error for KQL error. 
2. Please add Data connector schema at location .script/utils/schemas. You can refer to other scehmas for more clarification 
3. Add workbook metadata in the file Workbooks/WorkbooksMetadata.json 
4. Also add working images of workbook. 
5. Add input folder alias Data folder and solution metadata which is required to repackage the solution. You can refer to any other solution for more clarification

**Comment by v-prasadboke** [comment]
Hello @vkorenkov-varonis, Please add workbook metadata to the above mentioned file. Also add working images of workbook as well.

**Comment by vkorenkov-varonis** [comment]
@microsoft-github-policy-service agree [company="Varonis"]

**Comment by vkorenkov-varonis** [comment]
@microsoft-github-policy-service agree

**Comment by v-prasadboke** [comment]
Hello @vkorenkov-varonis, Thanks for committing the requested changes. Will take a look at it and come back to you by 18 January, 2024.
Thanks

File: Solutions/VaronisSaaS/Data Connectors/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    "_generator": {
+      "name": "bicep",
+      "version": "0.24.24.22086",
+      "templateHash": "1576663599375367683"
+    }
+  },
+  "parameters": {
+    "functionAppName": {
+      "type": "string",
+      "defaultValue": "[format('VaronisSaaS-{0}', uniqueString(resourceGroup().id))]",
```
Comment (v-prasadboke) [inline Solutions/VaronisSaaS/Data Connectors/azuredeploy.json:14]: Will it be okay if we let user add the custom Function app name, 
Also the zip and function app is missing with python libraries. Please add those as well.
You can refer this Data connector for more clarification [Solutions/MimecastTIRegional/Data Connectors](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/MimecastTIRegional/Data%20Connectors)

File: Solutions/VaronisSaaS/Data Connectors/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        "EP2",
+        "EP3"
+      ],
+      "metadata": {
+        "description": "Specifies the Azure Function hosting plan SKU."
+      }
+    },
+    "packageUri": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/VaronisSaaS/Data%20Connectors/Varonis.Sentinel.Functions.zip",
```
Comment (v-prasadboke) [inline Solutions/VaronisSaaS/Data Connectors/azuredeploy.json:81]: Also the default value should be a shortlink, I'll replace it later with aka.ms

File: Solutions/VaronisSaaS/Data/Solution_VaronisSaaS.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "Name": "VaronisSaaS",
+  "Author": "Varonis",
+  "Logo": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/VaronisLogo.svg\" width=\"75px\" height=\"75px\">",
+  "Description": "The Varonis SaaS integration allows you to retrieve Varonis DatAlert alerts, create incident and pull activities related to the alerts for conducting investigations.",
```
Comment (v-prasadboke) [inline Solutions/VaronisSaaS/Data/Solution_VaronisSaaS.json:5]: Please add a few more lines to the description

File: Solutions/VaronisSaaS/Data Connectors/VaronisSaaS_API_FunctionApp.json [hunk 4]
```diff
@@ -0,0 +1,127 @@
+{
+  "id": "VaronisSaaS",
+  "title": "Varonis SaaS",
+  "publisher": "Varonis",
+  "descriptionMarkdown": "Varonis SaaS provides the capability to ingest [Varonis Alerts](https://www.varonis.com/products/datalert) into Microsoft Sentinel.",
```
Comment (v-prasadboke) [inline Solutions/VaronisSaaS/Data Connectors/VaronisSaaS_API_FunctionApp.json:5]: Also add a few more lines to Description markdown of Data connector

File: Solutions/VaronisSaaS/Data Connectors/azuredeploy.json (hunk 1 above)
Comment (vkorenkov-varonis) [reply Solutions/VaronisSaaS/Data Connectors/azuredeploy.json:14]: Hi @v-prasadboke,
Currently user can change function name if he wants. It's just a default value.
Also we use .net as a platform for azure function. You can find source code and zip package in "Data Connectors" folder.

image
image

File: Solutions/VaronisSaaS/Data Connectors/VaronisSaaS_API_FunctionApp.json (hunk 4 above)
Comment (vkorenkov-varonis) [reply Solutions/VaronisSaaS/Data Connectors/VaronisSaaS_API_FunctionApp.json:5]: I'm waiting for copywriter, and I will update as soon as I have text.

**Comment by v-prasadboke** [comment]
Thank you for the updates @vkorenkov-varonis, Will check on it once and come back to you by 24 January, 2024.

File: Solutions/VaronisSaaS/Data Connectors/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+      "type": "securestring",
+      "metadata": {
+        "description": "Varonis DatAlert API key."
+      }
+    }
+  },
+  "variables": {
+    "hostingPlanName": "[parameters('functionAppName')]",
+    "applicationInsightsName": "[parameters('functionAppName')]",
+    "storageAccountName": "[format('{0}azfunctions', uniqueString(resourceGroup().id))]",
```
Comment (v-prasadboke) [inline Solutions/VaronisSaaS/Data Connectors/azuredeploy.json:115]: Hello @vkorenkov-varonis, replace azfunction with SA or limited character word. 
including azfunction is crossing the storagename limit. 
also can you share working images and output of function app.

Comment (vkorenkov-varonis) [reply Solutions/VaronisSaaS/Data Connectors/azuredeploy.json]: Renaming has been done. Output you can find here:
image

Not sure what do you mean by working images?
Workbook images can be found here:
image

**Comment by v-prasadboke** [comment]
Working images are function apps output results 
image

**Comment by vkorenkov-varonis** [comment]
@v-prasadboke 
image

**Comment by v-prasadboke** [comment]
Hello @vkorenkov-varonis we are all good with the solution. Just need a few more line to solution description.

File: Solutions/VaronisSaaS/Data Connectors/VaronisSaaS_API_FunctionApp.json (hunk 4 above)
Comment (v-prasadboke) [reply Solutions/VaronisSaaS/Data Connectors/VaronisSaaS_API_FunctionApp.json:5]: on this one

**Comment by v-prasadboke** [comment]
Hello @vkorenkov-varonis, Solution description needs a few more line. Please add them.

**Review by v-prasadboke — DISMISSED** [review]

**Comment by vkorenkov-varonis** [comment]
> Hello @vkorenkov-varonis, Solution description needs a few more line. Please add them.

waiting for copywriter

**Comment by v-prasadboke** [comment]
Hello @vkorenkov-varonis, Thanks for raising this PR. This PR will be investigated and will update you about the same before 01 February, 2024

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**